File: 1114-binary-search-tree-to-greater-sum-tree/binary-search-tree-to-greater-sum-tree.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */

int traverse(struct TreeNode *node, int sum) {
    if (node == NULL) {
        return sum;
    }

    sum = traverse(node->right, sum);

    node->val += sum;
    sum = node->val;

    return traverse(node->left, sum);
}

struct TreeNode* bstToGst(struct TreeNode* root) {
    traverse(root, 0);
    return root;
}

File: 1114-binary-search-tree-to-greater-sum-tree/binary-search-tree-to-greater-sum-tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def bstToGst(self, root):
        def traverse(node, sum):
            if not node:
                return sum
            
            x = traverse(node.right, sum)
            node.val += x
            y = traverse(node.left, node.val)
            return y
        
        traverse(root, 0)
        return root
